cc/src/main/scala/me/sgrouples/rogue/BsonFormats.scala

```diff
@@ -165,6 +165,18 @@ trait BaseBsonFormats {
     override def defaultValue: Double = 0.0d
   }
 
+  implicit object BigDecimalBsonFormat extends BasicBsonFormat[BigDecimal] {
+    override def read(b: BsonValue): BigDecimal = b.asDecimal128().decimal128Value().bigDecimalValue()
+    override def write(t: BigDecimal): BsonValue = new BsonDecimal128(new Decimal128(t.bigDecimal))
+    override def defaultValue: BigDecimal = BigDecimal(0)
+  }
+
+  implicit object CurrencyBsonFormat extends BasicBsonFormat[Currency] {
+    override def read(b: BsonValue): Currency = Currency.getInstance(b.asString().getValue)
+    override def write(t: Currency): BsonValue = new BsonString(t.getCurrencyCode)
+    override def defaultValue: Currency = Currency.getInstance(Locale.getDefault)
```

Comment:
@marekzebrowski Yeah, I had some thoughts about that, probably you're right, thou I don't know which should be the default and why USD ;) if we ever go open source with this for example...